src/main/java/org/opensearch/knn/index/codec/util/KNNCodecUtil.java

```diff
@@ -111,8 +114,45 @@ public static String buildEngineFilePrefix(String segmentName) {
         return String.format("%s_", segmentName);
     }
 
+    /**
+     * Build a suffix of file name with provided field name and custom extension.
+     * In case where the given field name contains invalid characters for the file name, it will escape them to '.'.
+     * Note that, field name MUST NOT contain '.' in it.
+     * <p>
+     * Ex:
+     * With 'my_vector' as field name and '.hnswc' as an extension -> _0_2011_my_vector.hnswc
+     * With 'my vector' (blank) as field name and '.hnswc' as an extension -> _0_2011_my.vector.hnswc
+     * With 'my/vector' (it has '/') as field name and '.hnswc' as an extension -> _0_2011_my.vector.hnswc
+     *
+     * @param fieldName : Vector field name
+     * @param extension : File extension.
+     * @return File suffix containing a field name that escaped with invalid characters.
+     */
     public static String buildEngineFileSuffix(String fieldName, String extension) {
```

Comment:
Should we keep backward compatibility? For example, with field name `test field`, we first look for `test.field` and if there is not such file we should look for `test field`.